vspd.go

```diff
@@ -0,0 +1,340 @@
+package dcrlibwallet
+
+import (
+	"bytes"
+	"context"
+	"crypto/ed25519"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/decred/dcrd/blockchain/stake/v3"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v2"
+	"github.com/decred/dcrd/dcrutil/v3"
+	"github.com/decred/dcrd/txscript/v3"
+	"github.com/decred/dcrd/wire"
+)
+
+type VSPD struct {
+	baseURL             string
+	pubKey              []byte
+	sourceWallet        *Wallet
+	mwRef               *MultiWallet
+	sourceAccountNumber int32
+	httpClient          http.Client
+	params              *chaincfg.Params
+}
+
+func (mw *MultiWallet) NewVSPD(baseURL string, sourceWallet *Wallet, sourceAccountNumber int32,
+	httpClient http.Client, params *chaincfg.Params) *VSPD {
+	return &VSPD{
+		baseURL:             baseURL,
+		sourceWallet:        sourceWallet,
+		mwRef:               mw,
+		sourceAccountNumber: sourceAccountNumber,
+		httpClient:          httpClient,
```

Comment:
For this construct
(*MultiWallet) has it own chainParams as well, we can just use from that
httpClient can create from here for more easy to integration